File: test/k6/irs.js
import { sleep } from "k6";
import http from "k6/http";

export const options = {
  vus: 10,
  duration: "30s",
};

export default function() {
  const idtrx = Date.now();
  const url = `http://127.0.0.1:8000/api/h2h?id=MA0001&pin=12345&user=MAU001&pass=12345&kodeproduk=SB5&tujuan=085123456789&counter=1&idtrx=${idtrx}&jenis=1`;
  http.get(url.toString());
  sleep(1);
}

File: internal/dto/irs.go
package dto

type IrsTransactionReq struct {
	ID         string `json:"id" query:"id"`
	PIN        string `json:"pin" query:"pin"`
	User       string `json:"user" query:"user"`
	Pass       string `json:"pass" query:"pass"`
	KodeProduk string `json:"kodeproduk" query:"kodeproduk"`
	Tujuan     string `json:"tujuan" query:"tujuan"`
	Counter    int    `json:"counter" query:"counter"`
	IDTrx      string `json:"idtrx" query:"idtrx"`
	Jenis      int    `json:"jenis" query:"jenis"`
}

type IrsTransactionRes struct {
	Success bool   `json:"success"`
	Produk  string `json:"produk"`
	Tujuan  string `json:"tujuan"`
	ReffID  string `json:"reffid"`
	RC      string `json:"rc"`
	Msg     string `json:"msg"`
}
